replica_aggregator/src/main/scala/uk/ac/wellcome/platform/storage/replica_aggregator/services/ReplicaAggregator.scala

```diff
@@ -0,0 +1,67 @@
+package uk.ac.wellcome.platform.storage.replica_aggregator.services
+
+import java.time.Instant
+
+import uk.ac.wellcome.platform.storage.replica_aggregator.models._
+import uk.ac.wellcome.storage.store.VersionedStore
+import uk.ac.wellcome.storage.{Identified, Version}
+
+import scala.util.Try
+
+class ReplicaAggregator(
+  versionedStore: VersionedStore[String, Int, Set[ReplicaResult]]
+) {
+  def aggregate(result: ReplicaResult): Try[ReplicationAggregationSummary] =
+    Try {
+
+      // When we add support for secondary storage locations, we need to tweak
+      // the logic for deciding if a replica is "complete" -- we'll need to
+      // check we have:
+      //
+      //    * at least one primary replica
+      //    * at least N secondary replicas, where N is configurable
+      //
+      // For the initial, simpler case, we just handle primary replicas, so
+      // error out if passed a secondary replica.
+      //
+      result.storageLocation match {
+        case _: SecondaryStorageLocation =>
+          throw new Throwable(
```

Comment:
Though I think this is interpreted as a deterministic error - it would be nicer to return a result type here indicative of that failure mode to be interpreted elsewhere. The intention is to replace this code path soon so no need to fix now.

Reply:
Yes, it's deliberately rubbish!

I wanted to put *something* here so if the pipeline goes wrong we see it, but it's coming out the next time we touch this code.